### ISSUE TITLE
Status of "save" or "edit" API?

### ISSUE THREAD

**Issue by kylewill** [internetarchive/openlibrary#4364]
The docs at https://openlibrary.org/dev/docs/restful_api reference a `save` API:

> Save
Modifying objects can be done by sending a PUT request to the resource url with appropriate Content-Type header. The currently supported format is only JSON.
>
>TODO: show an example
>
>Please note that right now this only an internal API and works only from the localhost.

The page hasn't been updated since 2015, is there a programmatic way to edit OpenLibrary `thing`s?

I'm building a simple background creator for zoom users that lets them populate their own books. 

https://backgroundstudio.app

The book cover portion is powered by OpenLibrary. Today I filter results where the book cover is not found. 

I want to add the functionality for users to submit covers and ideally contribute back to OL.

**Comment by cdrini** [comment]
Hi @kylewill ! Cool app :) Hmmm, if you're looking to upload a cover on their behalf (i.e. using _their_ Open Library account), that's currently not possible--we don't currently have something like OAuth for enabling that flow. If you're looking to upload covers programmatically using an account that _you_ own, then you can [create/register for a bot account](https://openlibrary.org/dev/docs/bots), after which those APIs will work.

**Comment by cdrini** [comment]
(You can also take a look at the python openlibrary-client to see the sort of requests it makes to e.g. upload a cover:)

https://github.com/internetarchive/openlibrary-client/blob/afd46d73daa85a64c2cd048bf91d3a50b68c8c43/olclient/openlibrary.py#L245-L252

**Comment by cdrini** [comment]
( This blog post does cheekily claim that it takes like "~10min" to setup, but I'm skeptical :smile: https://www.ory.sh/run-oauth2-server-open-source-api-security/ )

**Comment by kylewill** [comment]
thank you! The goal is to contribute book covers in general vs. on behalf of an author.

The use case is a user of my app searches for books to put on a virtual background. If there's not a book, I'd love to contribute the covers to OL.

One thing I've noticed, I added a book over a month ago and it still doesn't show up in the book cover API (or on the OL site). 

will file a separate issue for that. Thank you!
